starlite/utils/scope.py

```diff
@@ -52,9 +52,7 @@ def get_starlite_scope_state(scope: "Scope", key: str, default: Any = Empty) ->
         Value mapped to ``key`` in internal connection scope namespace.
     """
     namespace = scope["state"].setdefault(SCOPE_STATE_NAMESPACE, {})
-    if default is Empty:
-        return namespace.get(key)
-    return namespace.setdefault(key, default)
+    return namespace.pop(key, default if default is not Empty else None)
```

Comment:
I don't think that's right. Accessing a value should not delete it. If you call `get_starlite_scope_state(scope, "foo")` twice in a row, it should not return a different result both times.

Maybe just delete the value inside the middleware?

Either that or move this function into the `middleware/logigng.py`, since that's the only place its being used.

Reply:
I'm also not sure this fixes the issue, since overriding the key with an empty string doesn't seem to fix it either, which also results in the value not being referenced anymore in the state.

Reply:
i updated the implementation

Reply:
I'd still check with @chris-telemetry before merging this